src/cluster.c

```diff
@@ -2021,13 +2126,13 @@ void clusterProcessGossipSection(clusterMsg *hdr, clusterLink *link) {
                 !(flags & CLUSTER_NODE_NOADDR) &&
                 !(flags & (CLUSTER_NODE_FAIL|CLUSTER_NODE_PFAIL)) &&
                 (strcasecmp(node->ip,g->ip) ||
-                 node->port != ntohs(g->port) ||
+                 getNodeDefaultClientPort(node) != ntohs(g->port) ||
```

Comment:
do we need check `g->pport` here?

Reply:
I think this was more intended in the world where you are running a bunch of Redis processes on the same host but on different ports, maybe in a K8s cluster, so a pod may disappear and pop up on the same host but on a different port. 

You should probably check both port and pport, since we know port might not be used in the case we are trying to solve (cluster-tls no but using a tls port). We already derived the two ports, so you should be able to check them.